agent/bench-scripts/pbench-uperf

```diff
@@ -200,220 +201,220 @@ if [ $? -ne 0 ]; then
-		shift
-		if [ -n "$1" ]; then
-			runtime="$1"
-			shift
-		fi
+			if [ -n "$1" ]; then
+				runtime="$1"
+				shift
+			fi
 		;;
 		--log-response-times)
-		shift
-		if [ -n "$1" ]; then
-			log_response_times="$1"
-			shift
-		fi
+			if [ -n "$1" ]; then
+				log_response_times="$1"
+				shift
+			fi
 		;;
 		-p|--protocols)
-		shift
-		if [ -n "$1" ]; then
-			protocols="$1"
-			shift
-		fi
+			if [ -n "$1" ]; then
+				protocols="$1"
+				shift
+			fi
 		;;
 		-C|--client|--clients)
-		shift
-		if [ -n "$1" ]; then
-			clients="$1"
-			shift
-		fi
+			if [ -n "$1" ]; then
+				clients="$1"
+				shift
+			fi
 		;;
 		-S|--server|--servers)
-		shift
-		if [ -n "$1" ]; then
-			servers="$1"
-			shift
-		fi
+			if [ -n "$1" ]; then
+				servers="$1"
+				shift
+			fi
 		;;
 		--vsock-server|--vsock-servers)
-		shift
-		if [ -n "$1" ]; then
-			vsock_servers="$1"
-			shift
-		fi
+			if [ -n "$1" ]; then
+				vsock_servers="$1"
+				shift
+			fi
 		;;
 		-c|--config)
-		shift
-		if [ -n "$1" ]; then
-			config="$1"
-			shift
-		fi
+			if [ -n "$1" ]; then
+				config="$1"
+				shift
+			fi
 		;;
 		--start-iteration-num)
-		shift
-		if [ -n "$1" ]; then
-			start_iteration_num=$1
-			shift
-		fi
+			if [ -n "$1" ]; then
+				start_iteration_num=$1
+				shift
+			fi
 		;;
 		--client-node|--client-nodes)
-		shift
-		if [ -n "$1" ]; then
-			client_nodes="$1"
-			shift
-		fi
+			if [ -n "$1" ]; then
+				client_nodes="$1"
+				shift
+			fi
 		;;
 		--server-node|--server-nodes)
-		shift
-		if [ -n "$1" ]; then
-			server_nodes="$1"
-			shift
-		fi
+			if [ -n "$1" ]; then
+				server_nodes="$1"
+				shift
+			fi
 		;;
 		--sysinfo)
-		shift;
-		if [ -n "$1" ]; then
-			sysinfo="$1"
-			shift;
-		fi
+			if [ -n "$1" ]; then
+				sysinfo="$1"
+				shift
+			fi
 		;;
 		--)
-		shift
-		break
+			break
 		;;
 		*)
-		error_log "[$script_name] bad option, \"$1 $2\""
-		exit 1
-		break
+			error_log "[$script_name] bad option, \"${arg} ${1}\""
+			usage >&2
+			exit 1
 		;;
 	esac
 done
 verify_common_bench_script_options $tool_group $sysinfo
 
-# Verify the number of clients and servers match.
-if [ ! -z "$clients" -a "$postprocess_only" != "y" ]; then
+if [[ -z "${clients}" ]]; then
+	clients=${clients_def}
+fi
+
+if [[ -z "${servers}" ]]; then
+	servers=${servers_def}
+fi
+
+# before we run a test, verify clients and servers are accessible and install
+# uperf if necessary
+if [[ "${postprocess_only}" != "y" ]]; then
+	# Let's count how many localhost clients are servers are specified.
+	let check_local=0
+
+	# Verify the number of clients, servers, and option vsock servers match.
 	let c_cnt=0
-	for client in `echo $clients | sed -e s/,/" "/g`; do
-		let c_cnt=c_cnt+1
+	for client in ${clients//,/ }; do
+		(( c_cnt++ ))
+		if [[ "${client}" == "127.0.0.1" || "${client}" == "localhost" ]]; then
+			(( check_local++ ))
+		fi
 	done
+	if [[ ${c_cnt} -eq 0 ]]; then
+		# No clients on the command line, default to localhost.
+		c_cnt=1
+	fi
```

Comment:
Is this necessary? If `clients` was empty, you used `clients_def`, which should ensure it can never be empty here, so the loop should have always counted at least one.